volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import Callable, Tuple, List, Dict
+
+from volatility3.framework import renderers, interfaces, exceptions, constants, objects
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.objects import utility
+from volatility3.framework.symbols import linux
+from volatility3.plugins.linux import lsof
+
+
+vollog = logging.getLogger(__name__)
+
+class SockHandlers(interfaces.configuration.VersionableInterface):
+    """Handles several socket families extracting the sockets information."""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    def __init__(self, vmlinux, task):
+        self._vmlinux = vmlinux
+        self._task = task
+
+        netns_id = task.nsproxy.net_ns.get_inode()
+        self._netdevices = self._build_network_devices_map(netns_id)
+
+        self._sock_family_handlers = {
+            "AF_UNIX": self._unix_sock,
+            "AF_INET": self._inet_sock,
+            "AF_INET6": self._inet_sock,
+            "AF_NETLINK": self._netlink_sock,
+            "AF_VSOCK": self._vsock_sock,
+            "AF_PACKET": self._packet_sock,
+            "AF_XDP": self._xdp_sock,
+            "AF_BLUETOOTH": self._bluetooth_sock,
+        }
+
+    def _build_network_devices_map(self, netns_id: int) -> Dict:
+        """Given a namespace ID it returns a dictionary mapping each network
+        interface index (ifindex) to its network interface name:
+
+        Args:
+            netns_id: The network namespace ID
+
+        Returns:
+            netdevices_map: Mapping network interface index (ifindex) to network
+                            interface name
+        """
+        netdevices_map = {}
+        nethead = self._vmlinux.object_from_symbol(symbol_name="net_namespace_list")
+        net_symname = self._vmlinux.symbol_table_name + constants.BANG + "net"
+        for net in nethead.to_list(net_symname, "list"):
+            net_device_symname = self._vmlinux.symbol_table_name + constants.BANG + "net_device"
+            for net_dev in net.dev_base_head.to_list(net_device_symname, "dev_list"):
+                if net.get_inode() != netns_id:
+                    continue
```

Comment:
I'm not a huge fan of using continue, I find it makes following the flow more difficult.  It looks like this could easily be converted into `if net.get_inode() == netns_id:` and then the two statements and drop the need for the continue.